stem/output.py

```diff
@@ -0,0 +1,219 @@
+import warnings
+from abc import ABC, abstractmethod
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import List, Dict, Any
+from enum import Enum
+
+DEFAULT_GID_FLAGS = {
+    "WriteDeformedMeshFlag": "WriteUndeformed",
+    "WriteConditionsFlag": "WriteElementsOnly",
+    "GiDPostMode": "GiD_PostBinary",
+    "MultiFileFlag": "SingleFile",
+}
+
+
+class NodalOutput(Enum):
+    DISPLACEMENT = 1
+    DISPLACEMENT_X = 11
+    DISPLACEMENT_Y = 12
+    DISPLACEMENT_Z = 13
+    VELOCITY = 2
+    VELOCITY_X = 21
+    VELOCITY_Y = 22
+    VELOCITY_Z = 23
+    ACCELERATION = 3
+    ACCELERATION_X = 31
+    ACCELERATION_Y = 32
+    ACCELERATION_Z = 33
+    TOTAL_DISPLACEMENT = 4
+    TOTAL_DISPLACEMENT_X = 41
+    TOTAL_DISPLACEMENT_Y = 42
+    TOTAL_DISPLACEMENT_Z = 43
+    WATER_PRESSURE = 5
+    VOLUME_ACCELERATION = 6
+    VOLUME_ACCELERATION_X = 61
+    VOLUME_ACCELERATION_Y = 62
+    VOLUME_ACCELERATION_Z = 63
+
+
+TENSOR_OUTPUTS = [
+    "GREEN_LAGRANGE_STRAIN",
+    "ENGINEERING_STRAIN",
+    "CAUCHY_STRESS",
+    "TOTAL_STRESS",
+]
+
+
+def detect_tensor_outputs(requested_outputs: List[str]):
+    """
+    Detects whether gauss point outputs are requested and warns the user
```

Comment:
Detects whether tensor output on gauss points are requested and warns the user. This is required for VTK output and json output.